apitests/spec/devices_spec.ts

```diff
@@ -30,5 +30,96 @@ describe('devices api', () => {
                 done();
             });
         });
+
+        it('should always have these properties', (done) => {
+            request.get('', (err, resp, result) => {
+                expect(result.data[0].DeviceProperties).toBeTruthy();
+                expect(result.data[0].SystemProperties).toBeTruthy();
+                expect(result.data[0].Commands).toBeTruthy();
+                expect(result.data[0].CommandHistory).toBeTruthy();
+                expect(result.data[0].IsSimulatedDevice).toBeDefined();
+                expect(result.data[0].id).toBeTruthy();
+                expect(result.data[0]._rid).toBeTruthy();
+                expect(result.data[0]._self).toBeTruthy();
+                expect(result.data[0]._etag).toBeTruthy();
+                expect(result.data[0]._ts).toBeTruthy();
+                expect(result.data[0]._attachments).toBeTruthy();
+                done();
+            });
+        })
+
+        it('should have these attributes for enabled device', (done) => {
+            request.get('', (err, resp, result) => {
+                expect(result.data[1].DeviceProperties.HubEnabledState).not.toBeNull();
```

Comment:
This is going to be brittle. If the order changes, e.g. someone adds a new device, there's no guarantee result.data[1].DeviceProperties.HubEnabledState will not be null. You'll want to first do a find of any device in result.data that is hub enabled and then validate the properties below